// @tanstack/router-plugin/dist/cjs/ast.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const babel = require("@babel/core");
function _interopNamespaceDefault(e) {
  const n = Object.create(null, { [Symbol.toStringTag]: { value: "Module" } });
  if (e) {
    for (const k in e) {
      if (k !== "default") {
        const d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: () => e[k]
        });
      }
    }
  }
  n.default = e;
  return Object.freeze(n);
}
const babel__namespace = /* @__PURE__ */ _interopNamespaceDefault(babel);
function compileAst(makeOpts) {
  return async (opts) => {
    const res = babel__namespace.transformSync(opts.code, {
      plugins: [
        ["@babel/plugin-syntax-jsx", {}],
        [
          "@babel/plugin-syntax-typescript",
          {
            isTSX: true
          }
        ],
        ...opts.getBabelConfig().plugins
      ],
      root: makeOpts.root,
      filename: opts.filename,
      sourceMaps: true
    });
    if (res == null ? void 0 : res.code) {
      return {
        code: res.code,
        map: res.map
      };
    }
    return {
      code: opts.code,
      map: null
    };
  };
}
exports.compileAst = compileAst;
